invenio_deposit/api.py

```diff
@@ -131,12 +134,16 @@ def fetch_published(self):
         return resolver.resolve(pid_value)
 
     def merge_with_published(self):
-        """."""
-        # TODO if revisions are the same then we can apply directly
-        # latest = self.fetch_published()
-        # initial = A.revisions[self['_deposit']['pid']['revision']]
-        # patch = diff(initial, self)
-        # latest.apply(patch)
+        """Merge changes with latest published version."""
+        pid, first = self.fetch_published()
+        lca = first.revisions[self['_deposit']['pid']['revision_id']]
+        second = self
+        m = Merger(lca.dumps(), first.dumps(), second.dumps(), {})
```

Comment:
We should probably remove/ignore `_deposit` and `$schema` fields as they will be always different. The`_deposit` field should probably be set after merge like https://github.com/inveniosoftware/invenio-deposit/pull/63/files#diff-8d60a8f35673846c2c55cf5cde4c9b85R231.